### PR TITLE
Add checksum for py-astroid 2.12.7, py-astroid 2.12.10, py-setuptools 62.6.0, py-wrapt 1.14.1, py-pylint 2.15.0

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-astroid/package.py [hunk 1]
```diff
@@ -60,4 +64,5 @@ class PyAstroid(PythonPackage):
     depends_on("py-typing-extensions@3.7.4:", when="@2.7.3: ^python@:3.7", type=("build", "run"))
     depends_on("py-typing-extensions@3.10:", when="@2.8.3: ^python@:3.9", type=("build", "run"))
     depends_on("py-setuptools@17.1:", type=("build", "run"))
-    depends_on("py-setuptools@20:", type=("build", "run"), when="@2.7.3:")
+    depends_on("py-setuptools@20:", when="@2.7.3:", type=("build", "run"))
+    depends_on("py-setuptools@62.6:62", when="@2.12.7:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-astroid/package.py:69]: Need to lock down wheel versions too

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-astroid/package.py (hunk 1 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/py-astroid/package.py:69]: Done

**Review by adamjstewart — DISMISSED** [review]

**Comment by alalazo** [comment]
@spackbot run pipeline

**Comment by iarspider** [comment]
@spackbot run pipeline

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Comment by adamjstewart** [comment]
There will probably be conflicts between this PR and #32778

**Comment by iarspider** [comment]
> There will probably be conflicts between this PR and #32778

I have added the changes from that PR here and closed it.

**Review by adamjstewart — APPROVED** [review]

**Comment by iarspider** [comment]
Ah, I knew I have touched these packages recently! I guess I will update
the other PR with changes from this one, which you can close.

**Pull request merged**